variables.tf

```diff
@@ -1,36 +1,300 @@
+  type        = string
+  description = "Allowed networks for the Key Protect instance. Possible values: 'private-only', 'public-and-private'."
+  default     = "public-and-private"
+
+  validation {
+    condition     = can(regex("public-and-private|private-only", var.allowed_network))
+    error_message = "Valid values for allowed_network are 'public-and-private' or 'private-only'."
+  }
+}
+
+variable "standard_key" {
+  type        = bool
+  description = "Whether to create a standard key (true) or an imported key (false)."
+  default     = false
+}
+
+variable "endpoint_type" {
+  type        = string
+  description = "The type of endpoint for the KMS key. Options: 'public', 'private'."
+  default     = "public"
+}
+
+##############################################################################
+# COS bucket variables
+##############################################################################
+
+variable "bucket_name" {
+  type        = string
+  description = "The name for the new Object Storage bucket."
+  default     = "cybervault-bucket"
+}
+
+variable "bucket_storage_class" {
+  type        = string
+  description = "The storage class of the new bucket. Required only if `create_cos_bucket` is true. Possible values: `standard`, `vault`, `cold`, `smart`, `onerate_active`."
   default     = "standard"
+
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    condition     = can(regex("^standard$|^vault$|^cold$|^smart$|^onerate_active", var.bucket_storage_class))
+    error_message = "Variable 'bucket_storage_class' must be 'standard', 'vault', 'cold', 'smart' or 'onerate_active'."
   }
 }
 
-variable "resource_group_id" {
+variable "object_locking_enabled" {
+  description = "Whether to create an object lock configuration. Applies only if `object_versioning_enabled` and `create_cos_bucket` are true."
+  type        = bool
+  default     = false
+}
+
+variable "object_lock_duration_years" {
+  description = "The number of years for the object lock duration. If you specify a number of years, do not specify a value for `object_lock_duration_days`. Applies only if `create_cos_bucket` is set to `true`."
+  type        = number
+  default     = 1
+}
+
+variable "hard_quota" {
+  type        = number
+  description = "The hard quota (in GB) for the bucket. Set to 0 for unlimited."
+  default     = 1024
+}
+
+variable "force_delete" {
+  type        = bool
+  description = "Whether to force delete the key when deleting the resource."
+  default     = true
+}
+
+##############################################################################
+# cloud logs variable
+##############################################################################
+
+variable "cloud_log_instance_name" {
+  type        = string
+  description = "The name for the Cloud Logs Instance."
```

Comment:
this should mention that if user is passing a prefix, it will be added infront of this